core/store/src/db.rs

```diff
@@ -137,6 +137,12 @@ pub trait Database: Sync + Send {
     /// This is a no-op for in-memory databases.
     fn flush(&self) -> io::Result<()>;
 
+    /// Compact database representation.
+    ///
+    /// If the database supports it a form of compaction, calling this function
+    /// is blocking until compaction finishes. Otherwise, this is a no-op.
```

Comment:
```suggestion
    /// Compacts database representation.
    ///
    /// If the database supports compaction, calling this function blocks
    /// until compaction finishes. Otherwise, this is a no-op.
```